src/coreclr/src/jit/lsraxarch.cpp

```diff
@@ -1571,27 +1571,16 @@ int LinearScan::BuildPutArgStk(GenTreePutArgStk* putArgStk)
 
     ClassLayout* layout = src->AsObj()->GetLayout();
 
-    ssize_t size = putArgStk->gtNumSlots * TARGET_POINTER_SIZE;
+    ssize_t size = putArgStk->GetStackByteSize();
     switch (putArgStk->gtPutArgStkKind)
     {
         case GenTreePutArgStk::Kind::Push:
         case GenTreePutArgStk::Kind::PushAllSlots:
         case GenTreePutArgStk::Kind::Unroll:
             // If we have a remainder smaller than XMM_REGSIZE_BYTES, we need an integer temp reg.
-            //
-            // x86 specific note: if the size is odd, the last copy operation would be of size 1 byte.
-            // But on x86 only RBM_BYTE_REGS could be used as byte registers.  Therefore, exclude
-            // RBM_NON_BYTE_REGS from internal candidates.
             if (!layout->HasGCPtr() && (size & (XMM_REGSIZE_BYTES - 1)) != 0)
             {
                 regMaskTP regMask = allRegs(TYP_INT);
-
-#ifdef TARGET_X86
-                if ((size % 2) != 0)
```

Comment:
size was `putArgStk->gtNumSlots * TARGET_POINTER_SIZE` where `TARGET_POINTER_SIZE % 2 == 0` so that condition was always false.